io/disk/dlpar_vscsi.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Author: Bismruti Bidhibrata Pattjoshi<bbidhibr@in.ibm.com>
+
+"""
+DLPAR operations
+"""
+
+try:
+    import pxssh
+except ImportError:
+    from pexpect import pxssh
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+
+
+class CommandFailed(Exception):
+
+    '''
+    Defines the exception called when a
+    command fails
+    '''
+
+    def __init__(self, command, output, exitcode):
+        Exception.__init__(self, command, output, exitcode)
+        self.command = command
+        self.output = output
+        self.exitcode = exitcode
+
+    def __str__(self):
+        return "Command '%s' exited with %d.\nOutput:\n%s" \
+               % (self.command, self.exitcode, self.output)
+
+
+class DlparTest(Test):
+    '''
+    DLPAR disk script does vscsi device add,remove.
+    Update the details in yaml file.
+    '''
+
+    def setUp(self):
+        '''
+        Gather necessary test inputs.
+        '''
+        self.disk = self.params.get('disk', default=None)
+        self.vios_ip = self.params.get('vios_ip', '*', default=None)
+        self.vios_user = self.params.get('vios_username', '*', default=None)
+        self.vios_pwd = self.params.get('vios_pwd', '*', default=None)
+        self.login(self.vios_ip, self.vios_user, self.vios_pwd)
+        cmd = "lscfg -l %s" % self.disk
+        for line in process.system_output(cmd, shell=True).decode("utf-8") \
+                                                          .splitlines():
+            if self.disk in line:
+                self.slot = line.split()[-1].split('-')[-2]
+        cmd = "lsdev -slots"
+        output = self.run_command(cmd)
+        print(output)
+        for line in output.splitlines():
+            if self.slot in line:
+                self.host = line.split()[-1]
+        self.log.info(self.host)
+        cmd = "lsscsi -vl | tr -d '[]'"
+        output = process.system_output(cmd, shell=True)
+        for line in output.decode("utf-8").splitlines():
+            if self.disk in line:
+                value = line.split()[0]
+        for line in output.decode("utf-8").splitlines():
+            if value in line:
+                if "dir" in line:
+                    self.disk_dir = line.split()[-1]
+        cmd = r"cat %s/inquiry" % self.disk_dir
+        output = process.system_output(cmd, shell=True)
+        self.hdisk_name = output.split()[2].strip(b'0001').decode("utf-8")
```

Comment:
It can be decoded to utf-8 and then 0001 can be stripped

Reply:
if i will put decode("utf-8") "output = process.system_output(cmd, shell=True)" here getting error, so put after that.

Reply:
oh ok